website_portal_purchase/controllers/main.py

```diff
@@ -0,0 +1,66 @@
+# -*- coding: utf-8 -*-
+# (c) 2015 Antiun Ingeniería S.L. - Sergio Teruel
+# (c) 2015 Antiun Ingeniería S.L. - Carlos Dauden
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+
+import datetime
+
+from openerp import http
+from openerp.http import request
+
+from openerp.addons.website_portal.controllers.main import WebsiteAccount
+
+
+class PortalPurchaseWebsiteAccount(WebsiteAccount):
+
+    @http.route(['/my/supplier/home'], type='http', auth="user", website=True)
+    def supplier_account(self, **kw):
+        """ Add purchase documents to main account page """
+        response = super(PortalPurchaseWebsiteAccount, self).account(**kw)
+
+        partner = request.env.user.partner_id
+        quotations = request.env['purchase.order'].search([
+            ('partner_id.id', '=', partner.id),
+            ('state', 'in', ['sent', 'bid', 'confirmed', 'cancel'])
+        ])
+        orders = request.env['purchase.order'].search([
+            ('partner_id.id', '=', partner.id),
+            ('state', 'in', ['approved', 'except_picking',
+                             'except_invoice', 'done'])
+        ])
+        invoices = request.env['account.invoice'].search([
+            ('partner_id.id', '=', partner.id),
+            ('state', 'in', ['open', 'paid', 'cancelled']),
+            ('type', 'in', ['in_invoice', 'in_refund'])
+        ])
+
+        response.qcontext.update({
+            'date': datetime.date.today().strftime('%Y-%m-%d'),
+            'quotations': quotations,
+            'orders': orders,
+            'invoices': invoices,
+        })
+        return response
+
+    @http.route(['/my/supplier/orders/<int:order>'], type='http', auth="user",
+                website=True)
+    def supplier_orders_followup(self, order=None):
+        partner = request.env.user.partner_id
+        domain = [
+            ('partner_id.id', '=', partner.id),
+            ('state', 'not in', ['draft', 'cancel']),
+            ('id', '=', order)
+        ]
+        order = request.env['purchase.order'].search(domain)
+        invoiced_lines = request.env['account.invoice.line'].search([
+            ('invoice_id', 'in', order.invoice_ids.ids)
+        ])
+        order_invoice_lines = {il.product_id.id: il.invoice_id
+                               for il in invoiced_lines}
+
+        return request.website.render(
+            "website_portal_purchase.orders_followup",
+            {
+                'order': order.sudo(),
```

Comment:
Why sudo here? This is too dangerous. You should adjust security for portal users, but not allow to access any linked record